streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoinTest.java

```diff
@@ -84,6 +90,134 @@ public void shouldLogAndMeterOnSkippedRecordsWithNullValue() {
         }
     }
 
+    @Test
+    public void shouldThrowExceptionThisStoreSupplierRetentionDoNotMatchWindowsSizeAndGrace() {
```

Comment:
This should be multiple test. One for each case.

Reply:
Ack